sdk/vppagent/acl.go

```
@@ -0,0 +1,319 @@
+package vppagent
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"strconv"
+	"strings"
+
+	"github.com/golang/protobuf/ptypes/empty"
+	"github.com/ligato/vpp-agent/api/configurator"
+	"github.com/ligato/vpp-agent/api/models/vpp"
+	acl "github.com/ligato/vpp-agent/api/models/vpp/acl"
+	"github.com/networkservicemesh/networkservicemesh/controlplane/pkg/apis/local/connection"
+	"github.com/networkservicemesh/networkservicemesh/controlplane/pkg/apis/local/networkservice"
+	"github.com/networkservicemesh/networkservicemesh/pkg/tools"
+	"github.com/networkservicemesh/networkservicemesh/sdk/common"
+	"github.com/networkservicemesh/networkservicemesh/sdk/endpoint"
+	"github.com/sirupsen/logrus"
+)
+
+// ACL is a VPP Agent ACL composite
+type ACL struct {
+	endpoint.BaseCompositeEndpoint
+	Rules       map[string]string
+	Connections map[string]*ConnectionData
+}
+
+// Request implements the request handler
+func (a *ACL) Request(ctx context.Context, request *networkservice.NetworkServiceRequest) (*connection.Connection, error) {
+
+	if a.GetNext() == nil {
+		logrus.Fatal("The VPP Agent ACL composite requires that there is Next set")
```

Should we return an error here?

---

Fixed